### DISCUSSION TITLE
2023.32.1 feedback MCU2 2013 MS - BIG improvements!

### DISCUSSION THREAD

**Discussion by ratepOG** [tesla-android/issue-tracker#291] [General]
- **Installed the 2023.32.1 update via the Android OS updater** - worked flawlessly, took some time after download, maybe 30+ minutes, which was expected

- **GPS integration** with the Tesla browser works well enough with Waze
   - The update interval from the Tesla browser must be variable, Waze displays between 10 MPH under and over the actual speed every couple of seconds, instead of the actual speed. Not a deal breaker, but can be annoying if you enable the over speed limit warning in Waze.
   - The tablet version of Waze seems to be different with voice prompts versus the mobile app, just and interesting observation. The tablet version isn't specific, just "hazard ahead" instead vehicle stopped, object on road, etc.
   - Split screen is handy for Waze and an audio app, but it's not as polished as Android Auto, but acceptable

- **Audio playback in the browser**, the release notes say:
> Playback is allowed even when Drive or Reverse is engaged, meaning that there is no need to pair Tesla Android with your car using Bluetooth

however on a 2013 Model S the audio drops immediately when I go into reverse and the backup camera comes on. The browser relaunch starts the browser app up as if it wasn't opened and you need to tap the device.teslaandroid.com link. Probably an MCU2 memory limitation and the app is unloaded. Annoying, but workable. Maybe I'll switch to bluetooth, but that would remove the ability to take calls, which would not be preferrable, since only one device can be connected the the Tesla bluetooth service.
- Any thoughts on connecting the phone to the Android OS and routing calls through there? Not sure if that's possible, what mic would be used, etc., but that would be interesting.
   - Along the same thought, it would be nice to be able to see and interact with texts from the phone in the Android tablet interface. Starting to get into Android Auto features, but I figured I'd ask.

- **Tesla App switching** - when you got to another Tesla app (such as the power usage graph) the audio cuts out immediately, maybe same issue as above. 
   - When you reconnect to the Android OS in the browser the screen comes up to where you were before, usually Waze, but the audio takes a long time to come back. It seems that it comes back when I click the square at the bottom of the screen. I don't have to click the app that is providing the audio, it just comes back, then I click Waze again.

This update makes things much more usable. Maybe some of this could be addressed in future improvements. I'm hoping to avoid needing to use Android Auto with the Carlinkit dongle.

**Comment by drderiv** [comment]
Agreed, BIG improvements!  With respect to using Bluetooth, which I prefer because you can interact with the screen to do anything (other than a phone call, as mentioned) and your TA audio continues uninterrupted, with 2023.32.1 in my 2022 Model Y I am getting double audio, i.e., BT audio as well as browser audio, even with the browser audio switch set to disabled.  I can test it more, but my recollection was that it wasn't doubled all the time, just under certain scenarios.  But once it started doubling, going to the switch, enabling and disabling didn't fix it.  I'd be curious to hear if anyone else has had this happen.

**Reply by mikegapinski** [reply]
The switch for browser audio is broken 😅 I will push a hotfix release soon

**Reply by maxicusj** [reply]
in the latest 40 version I got only bluetooth audio. is this audio switch some sort of button that can be triggered manually?

**Comment by drderiv** [comment]
Same issue with the audio switch for me in 2023.32.2.  Still getting double audio when the Pi is connected to the car via Bluetooth.

**Comment by mikegapinski** [comment]
Did you clear the browser data after the update?

Wysyłane z aplikacji Outlook dla systemu iOS<https://aka.ms/o0ukef>
________________________________
Od: drderiv ***@***.***>
Wysłane: Thursday, August 31, 2023 1:45:43 AM
Do: tesla-android/issue-tracker ***@***.***>
DW: Michał Gapiński ***@***.***>; Comment ***@***.***>
Temat: Re: [tesla-android/issue-tracker] 2023.32.1 feedback MCU2 2013 MS - BIG improvements! (Discussion #291)


Same issue with the audio switch for me in 2023.32.2. Still getting double audio when the Pi is connected to the car via Bluetooth.

**Reply by drderiv** [reply]
No, I forgot that step, and thank you for the reminder.  I just tacked on a random query string to the TA URL, e.g., ?a=123 and that seemed to force a script reload which fixed the issue.

**Reply by drderiv** [reply]
I just installed 2023.36.1 and now I'm back to doubled audio when connected on BT regardless of the browser audio switch position.  And this time, I did a full clear of the browser data in the Service menu, then rebooted the Pi, and still, doubled audio.  Anything else I might be doing wrong?
